### PR TITLE
Fix index error msg

### LINKED ISSUE #202: [PE-D][Tester C] view: error message for hugely negative index isn't specific enough
When I try `view -2147483649`, I get the error:

```
Oops! The index you provided is not valid.

It appears to be either not a valid integer or it exceeds the maximum allowed value (2147483647).

Please try again with a valid positive integer!
```

I would expect it to have the same error as `view -10`, which is more accurate/specific:

```
Oops! The index you have given is not a positive integer. Please try again with a positive integer!
```

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: kokrui/ped#7

### PR DESCRIPTION
**Pull request by papataco14**
Fix negative index error message for extreme inputs to be correct mathematically and logically.

Closes #202

### CODE REVIEW

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -26,9 +26,9 @@ public class Messages {
                 "Oops! Multiple values are specified for the following single-valued parameter(s): ";
 
     public static final String MESSAGE_NON_INTEGER_INDEX = "Oops! The index you provided is not valid.\n"
-            + "It appears to be either not a valid integer or it exceeds the maximum allowed value "
-            + "(2147483647).\n"
-            + "Please try again with a valid positive integer!";
+            + "It should be a positive integer and within the range of 1 to "
+            + Integer.MAX_VALUE + ".\n"
+            + "Please try again with a valid index!";
```
Comment (tayruxin) [inline src/main/java/seedu/address/logic/Messages.java:31]: Should we state what is Integer.MAX_VALUE?

Comment (papataco14) [reply src/main/java/seedu/address/logic/Messages.java:31]: That is just how it shows in the code, the actual value is displayed to the user

File: src/main/java/seedu/address/logic/Messages.java [hunk 2]
```diff
@@ -26,9 +26,9 @@ public class Messages {
...
 
     public static final String MESSAGE_NON_POSITIVE_INTEGER_INDEX = "Oops! The index you have given is not a "
```
Comment (AlagappanRa) [inline src/main/java/seedu/address/logic/Messages.java:33]: Are you sure this gives the correct output to the user, when the user types a number > Integer.MAX_VALUE?

File: src/main/java/seedu/address/logic/Messages.java [hunk 3]
```diff
@@ -26,9 +26,9 @@ public class Messages {
                 "Oops! Multiple values are specified for the following single-valued parameter(s): ";
 
     public static final String MESSAGE_NON_INTEGER_INDEX = "Oops! The index you provided is not valid.\n"
-            + "It appears to be either not a valid integer or it exceeds the maximum allowed value "
-            + "(2147483647).\n"
-            + "Please try again with a valid positive integer!";
+            + "It should be a positive integer and within the range of 1 to "
```
Comment (AlagappanRa) [inline src/main/java/seedu/address/logic/Messages.java:29]: The case of not an integer and the case for not in range from 1 to Integer.MAX_VALUE should be split into two seperate cases.

**Review by AlagappanRa — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/Messages.java (hunk 3 above)
Comment (AlagappanRa) [reply src/main/java/seedu/address/logic/Messages.java:29]: Not an integer check should take precedence over not in range check

Comment (AlagappanRa) [reply src/main/java/seedu/address/logic/Messages.java:29]: Also what are we doing for these cases:
https://byjus.com/us/math/the-different-forms-of-representing-numbers/#:~:text=The%20three%20forms%20to%20represent,%2C%20500%20%2B%2020%20%2B%203.

1. `100          000` [Spaces format]
2. `100,000` [Standard Form] 
3.  `three hundred ninety-nine` [Word Form]

Comment (papataco14) [reply src/main/java/seedu/address/logic/Messages.java:29]: > The case of not an integer and the case for not in range from 1 to Integer.MAX_VALUE should be split into two seperate cases.
> 
> Possible to get flagged for potential feature flaw for non-specific error message.

When it comes to telling the user what went wrong with their input index, distinguishing between negative and nonstring indexes provides no additional value.

Comment (AlagappanRa) [reply src/main/java/seedu/address/logic/Messages.java:29]: Alright if you say so, will approve for now ...

**Review by AlagappanRa — APPROVED** [review]

**Pull request merged**